app/lib/log_util.rb

```diff
@@ -1,10 +1,16 @@
+# frozen_string_literal: true
+
 class LogUtil
-  def self.log_err_and_airbrake(msg)
-    Rails.logger.error(msg)
+  def self.log_err_and_airbrake(msg, exception = nil)
```

Comment:
Either exception should not be nil by default or we should check for nil below... 
Weird that it is optional here and required below.

Reply:
I agree it looks weird and I felt it too. However, I wanted to keep the previous signatures and just extend their behavior.
These functions have different purposes, which explain the order of these parameters.

`log_err_and_airbrake` sends a *string* message (`msg`) to airbrake. It doesn't accept an exception. For me the name of this operation is wrong, since we are just sending a string to airbrake, not necessarily an error.

So in order to keep it compatible to the current implementation, I added the `exception` field as optional (otherwise it would require a big refactoring to add this second parameter in dozens of different places)

Reply:
As discussed, my preferred approach is to create a new function with the ideal behavior and mark the other one as deprecated.
The new function should be marked has preferred - maybe with some comment to justify it. This will drive adoption of the recommended function as opposed to add some confusion to the current codebase by trying to be backwards compatible. 
We can also refactor `log_err_and_airbrake` to use the new functions...

Reply:
I believe the right approach would be wrapping the original exception in more specific exception and sending it directly to airbrake. Since this is a huge refactoring, I opted for reverting the method to its original format and improving the `log_backtrace` to support an additional message.

Reply:
I've completely reverted the error logging to its original form, since this change is completely unrelated to this PR. I missed that extra bit of info while I was making this change, but doing it "right" will require changing multiple places.
We can create a task later on to address a better error handling pattern.